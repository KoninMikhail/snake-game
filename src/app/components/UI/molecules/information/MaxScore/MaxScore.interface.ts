import { TextAlign } from '@/types/typography';
import { AssertPositive } from '@/types/assert-positive';

export interface IMaxScore<T extends AssertPositive<number>> {
    value: T;
    label: string;
    align?: TextAlign;
}
