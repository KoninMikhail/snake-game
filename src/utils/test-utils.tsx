import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
