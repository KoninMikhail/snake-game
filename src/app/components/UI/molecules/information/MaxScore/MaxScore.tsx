import {Caption} from '@ui/atoms/typography/caption/Caption/Caption';
import {IMaxScore} from '@ui/molecules/information/MaxScore/MaxScore.interface';

/**
 * @name MaxScore
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use it for display count engine score;
 *
 * @param value - score from engine
 * @param label - label text
 * @param align - text alignment
 * @return {JSX.Element}
 */
export const MaxScore = <T extends number>({ value, label, align = 'left' }: IMaxScore<T>) => (
    <Caption align={align}>{label + ' ' + value}</Caption>
);
