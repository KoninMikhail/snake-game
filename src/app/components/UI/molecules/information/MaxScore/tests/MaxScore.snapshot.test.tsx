import {MaxScore} from '../MaxScore';
import {renderWithProviders} from '@utils/test-utils';

/**
 * Tests
 */
describe('Score', () => {
    const label = 'record:';
    const value = 30;

    it('equal snapshot', () => {
        const {container} = renderWithProviders(<MaxScore value={value} label={label}/>);
        expect(container).toMatchSnapshot();
    });
});
