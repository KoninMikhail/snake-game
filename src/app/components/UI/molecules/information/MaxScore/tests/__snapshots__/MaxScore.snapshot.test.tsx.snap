// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score equal snapshot 1`] = `
.c0 {
  font-size: 1.3em;
  font-weight: normal;
  text-align: left;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1.2em;
  }
}

@media (min-width:1200px) {
  .c0 {
    font-size: 1.18em;
  }
}

<div>
  <span
    class="c0"
  >
    record: 30
  </span>
</div>
`;
